### PR TITLE
Adding VARCHAR type to mini-runners and random fixes

### PR DESCRIPTION
**Pull request by malin1993ml**
This PR mainly augments the mini-runners to support the VARCHAR type for sequential scans and index scans. For now, all the varchars are inlined because dealing with memory management with non-inlined varchars requires additional engineering.

Other random changes/fixes:
1. Extend the bwtree index template to support 512-byte key.
2. Extend the global trainer to support data collected with sampling.
3. Fix the data size calculation for `SEQ_SCAN` in `operating_unit_recorder`.
4. Adjust the key size calculation for VARCHARs in `operating_unit_recorder`.

### CODE REVIEW

**Review by 17zhangw — CHANGES_REQUESTED** [review]
Generally LGTM! Just some minor clarification on whether some changes should be committed.

File: src/include/main/db_main.h [hunk 1]
```diff
@@ -722,7 +722,7 @@ class DBMain {
      */
     std::unique_ptr<metrics::MetricsManager> BootstrapMetricsManager() {
       std::unique_ptr<metrics::MetricsManager> metrics_manager = std::make_unique<metrics::MetricsManager>();
-      if (metrics_pipeline_) metrics_manager->EnableMetric(metrics::MetricsComponent::EXECUTION_PIPELINE, 0);
+      if (metrics_pipeline_) metrics_manager->EnableMetric(metrics::MetricsComponent::EXECUTION_PIPELINE, 9);
```
[the remark this answers is not in the document]
Comment (malin1993ml) [reply src/include/main/db_main.h:725]: I think let's do this. This avoids the metrics collection having too much overhead to affect the system behavior itself, especially in concurrent OLTP workloads. I'm using this even in the single-threaded setting now to get a consistent measurement, though the prediction accuracy looks similar to recording everything.

File: script/model/mini_trainer.py [hunk 2]
```diff
@@ -164,11 +164,11 @@ def train(self):
 # ==============================================
 if __name__ == '__main__':
     aparser = argparse.ArgumentParser(description='Mini Trainer')
-    aparser.add_argument('--input_path', default='mini_runner_input',
+    aparser.add_argument('--input_path', default='mini_runner_input_varchar',
                          help='Input file path for the mini runners')
     aparser.add_argument('--model_results_path', default='mini_runner_model_results',
                          help='Prediction results of the mini models')
-    aparser.add_argument('--save_path', default='trained_model', help='Path to save the mini models')
+    aparser.add_argument('--save_path', default='trained_model_varchar', help='Path to save the mini models')
```
[the remark this answers is not in the document]
Comment (malin1993ml) [reply script/model/mini_trainer.py]: Those paths are a bit random anyway... but you're correct that the changes are unnecessary.

**Review by 17zhangw — APPROVED** [review]
LGTM

**Pull request merged**